fintrack/src/components/TransactionForm.js

```diff
@@ -0,0 +1,208 @@
+import React, { useState, useContext, useEffect } from 'react'
+import { useNavigate, useLocation } from 'react-router-dom'
+
+import { createTransaction, updateTransaction, deleteTransaction } from '../utils/api-transactions'
+
+import UserContext from '../context/UserContext'
+import { AuthContext } from '../context/AuthContext'
+
+function TransactionForm() {
+    const navigate = useNavigate()
+    const location = useLocation()
+
+    const { authToken } = useContext(AuthContext)
+    // get the user's categories and transaction types from context
+    const { authUser } = useContext(UserContext)
+
+    const [isEditMode, setIsEditMode] = useState(false)
+    const [account, setAccount] = useState(location.state?.selectedAccount)
+    const [transaction, setTransaction] = useState(location.state.transaction)
+    const [formData, setFormData] = useState({
+        account: '',
+        transactionType: '',
+        category: '',
+        amount: '',
+        description: ''
+    })
+    const [feedbackMessage, setFeedbackMessage] = useState('')
+
+    useEffect(() => {
+        console.log('use effect')
```

Comment:
When it comes to logging, I think there is a ton of value for leaving some logging in production. However, to be a bit more professional about it, ideally there should be some global switch for logging to be turned on. It should be turned on in development and off in production by default, and it's very very sweet if there's an easy way to turn that logging instrumentation on in production easily.

For a client-side app, this is pretty simple!

function log(msg) {
  if (localStorage.getItem('loggerEnabled') || 'localhost' in window.location.origin) {
    console.log(msg);
  }
}

So, we use `log` in our app instead of `console.log`.

* it's turned on by default if the app is running on a localhost domain
* it's off by default in prod
* if we go in the dev tools and change localStorage in prod, we can open this secret hatch and get logging to happen in prod!